ghost/core/core/server/services/mega/post-email-serializer.js

```diff
@@ -297,18 +321,87 @@ const serialize = async (postModel, newsletter, options = {isBrowserPreview: fal
         html: formatHtmlForEmail(htmlTemplate),
         plaintext: post.plaintext
     });
-
-    return {
+    const data = {
         subject: post.email_subject || post.title,
         html,
         plaintext
     };
+    if (labs.isSet('newsletterPaywall')) {
+        data.post = post;
+    }
+    return data;
 };
 
+/**
+ * renderPaywallCTA
+ *
+ * outputs html for rendering paywall CTA in newsletter
+ *
+ * @param {Object} post Post Object
+ */
+
+function renderPaywallCTA(post) {
+    const accentColor = settingsCache.get('accent_color');
+    const siteTitle = settingsCache.get('title') || 'Ghost';
+    const signupUrl = createPostSignupUrl(post);
+
+    return `<div class="align-center" style="text-align: center;">
```

Comment:
Longer term, we're going to need a better way to do this - ideally handlebars and partials I think. Defo fine for now though

Reply:
Yeah agree 💯